java-manta-it/src/test/java/com/joyent/manta/client/MantaClientIT.java

```diff
@@ -10,69 +10,51 @@
 import com.joyent.manta.client.helper.IntegrationTestHelper;
 import com.joyent.manta.config.ConfigContext;
 import com.joyent.manta.config.IntegrationTestConfigContext;
-import com.joyent.manta.exception.MantaClientException;
-import com.joyent.manta.exception.MantaClientHttpResponseException;
-import com.joyent.manta.exception.MantaErrorCode;
-import com.joyent.manta.exception.MantaUnexpectedObjectTypeException;
 import com.joyent.manta.http.MantaHttpHeaders;
-import com.joyent.manta.util.MantaUtils;
 import com.joyent.test.util.MantaAssert;
 import com.joyent.test.util.MantaFunction;
 import com.joyent.test.util.RandomInputStream;
 import org.apache.commons.io.FileUtils;
 import org.apache.commons.io.IOUtils;
 import org.testng.Assert;
-import org.testng.annotations.AfterClass;
-import org.testng.annotations.BeforeClass;
-import org.testng.annotations.Optional;
-import org.testng.annotations.Parameters;
-import org.testng.annotations.Test;
+import org.testng.annotations.*;
 
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.nio.charset.Charset;
 import java.nio.charset.StandardCharsets;
-import java.nio.file.Files;
-import java.util.Date;
 import java.util.UUID;
 import java.util.concurrent.Callable;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
-import java.util.concurrent.atomic.AtomicInteger;
-import java.util.stream.Stream;
 
-import static com.joyent.manta.client.MantaClient.SEPARATOR;
 import static com.joyent.manta.exception.MantaErrorCode.RESOURCE_NOT_FOUND_ERROR;
 
 /**
  * Tests the basic functionality of the {@link MantaClient} class.
  */
-@Test(groups = {"encrypted"})
+@Test(groups = {"buckets"})
 public class MantaClientIT {
 
     private static final String TEST_DATA = "EPISODEII_IS_BEST_EPISODE";
 
-    private final MantaClient mantaClient;
+    private MantaClient mantaClient;
 
-    private final String testPathPrefix;
+    private String testPathPrefix;
 
+    @BeforeClass
     @Parameters({"encryptionCipher", "testType"})
-    public MantaClientIT(final @Optional String encryptionCipher, final @Optional String testType) throws IOException {
+    public void beforeClass(final @Optional String encryptionCipher, final @Optional String testType) throws IOException {
 
         // Let TestNG configuration take precedence over environment variables
         ConfigContext config = new IntegrationTestConfigContext(encryptionCipher);
-        String testName = this.getClass().getSimpleName();
+        final String testName = this.getClass().getSimpleName();
 
         mantaClient = new MantaClient(config);
         testPathPrefix = IntegrationTestHelper.setupTestPath(config, mantaClient,
```

Comment:
The `testPathPrefix` instantiated here does not have a `MantaClient.SEPARATOR` as a trailing character, so when you run tests like `canReadStreamAndThenCloseWithoutErrors()`, you will see an oddly concatenated path like: `/elijah.zupancic/buckets/java-integration-tests/objects/MantaClientITff279399-3a21-42d1-897c-9b2457dab20f`

There needs to be a separator character between the test name and the UUID.

Reply:
May be a hyphen as a separator.

Reply:
Object Names now allow SEPARATOR. So the Integration-test base path will be reconfigured accordingly as : `ashwin.nair/buckets/java-integration-tests/objects/MantaClientIT/ff279399-3a21-42d1-897c-9b2457dab20f`